### DISCUSSION TITLE
Postgres Support

### DISCUSSION THREAD

**Discussion by jgkawell** [nkanaev/yarr#188] [Show and tell]
The author of `yarr` has already made it clear [here](https://github.com/nkanaev/yarr/issues/57) that the application is feature complete so this issue is not me asking for this support to be added but instead is just a note to anyone else in my position on how to get this functionality.

I wanted a simple, self-hosted RSS feed manager/reader so that I could both own my data and sync my RSS feeds between all my devices. I found `yarr` and it was perfect except for its dependency on sqlite. I typically run my self-hosted services in Kubernetes and like to reduce the number of services that are stateful. To do this I run a PostgreSQL instance and connect most of my services to it so that I can centralize backups and maintenance of the persistence layer.

To this end, I [forked yarr](https://github.com/jgkawell/yarr) and removed both the sqlite dependency as well as all the bits that were used exclusively for running it as a local application. If you also want PostgreSQL support feel free to check out my fork and use it/fork it yourself as you wish. I have Docker images being published from it [here](https://hub.docker.com/r/jgkawell/yarr) which you can use as is if you don't want to build it yourself.

One final note: if you want support for other databases (MySQL, MariaDB, etc.) please open an issue or a PR on my fork and I'll take a look at making that possible. I'm using the `pgx/stdlib` driver so making the driver swappable for other databases should be fairly straightforward.

Links:
- fork with postgres support: https://github.com/jgkawell/yarr
- docker images with postgres support: https://hub.docker.com/r/jgkawell/yarr

**Comment by nkanaev** [comment]
Postgres support is upcoming in v2.7

**Comment by nkanaev** [comment]
Docker image supporting Postgres has been published to Docker Hub. See the links below for more info:

* https://nkanaev.github.io/yarr/en/installation/docker/
* https://nkanaev.github.io/yarr/en/configuration/storage/
